flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -70,10 +77,41 @@
 	private static final Pattern PATTERN_NESTED_FIELDS = Pattern.compile(REGEX_NESTED_FIELDS);
 	private static final Pattern PATTERN_NESTED_FIELDS_WILDCARD = Pattern.compile(REGEX_NESTED_FIELDS_WILDCARD);
 
+	private static final Map<Class<?>, Class<? extends TypeSerializer>> customSerializers = new HashMap<>();
+	private static final Map<Tuple2<ArrayList<Integer>, Class>, Class<? extends TypeComparator>> customComparators =
+		new HashMap<>();
+
 	private final PojoField[] fields;
 	
 	private final int totalFields;
 
+	/**
+	 * Register a custom serializer for a type. The precedence of the serializers
+	 * is the following (highest to lowest): Kryo, Avro, Custom, Generated, Flink.
+	 * The chosen serializer will be the first one from the list that is turned on.
+	 *
+	 */
+	public static <C, S extends TypeSerializer<C>> void registerCustomSerializer(Class<C> clazz, Class<S> ser) {
+		Constructor<?>[] ctors = ser.getConstructors();
+		assert ctors.length == 1;
+		assert ctors[0].getParameterTypes().length == 0;
```

Comment:
Is it documented somewhere that custom serializers have to have these properties?

Also, these shouldn't be asserts, but throw exceptions instead. (I think asserts are generally for internal consistency stuff, i.e., they should fire only when you have a bug, and not when some stuff that a user gave us is not in the right form.)